### PR TITLE
Add ERC: ERC-20 Holder Extension for NFTs

### PR DESCRIPTION
**Pull request by steven2308**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc20-holder-extension-for-nfts.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+---
```
Comment (abcoathup) [inline ERCS/erc20-holder-extension-for-nfts.md:1]:
```suggestion
---
eip: 7590
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

Comment (steven2308) [reply ERCS/erc20-holder-extension-for-nfts.md]: Perfect, thanks!

File: ERCS/erc-7590.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-05
+requires: 20, 165, 721
+---
+
+## Abstract
+
+This proposal suggests an extension to [ERC-721](./eip-721.md) to enable easy exchange of [ERC-20](./eip-20.md) tokens. By enhancing [ERC-721](./eip-721.md), it allows unique tokens to manage and trade [ERC-20](./eip-20.md) fungible tokens bundled in a single NFT.
```
Comment (SamWilsn) [inline ERCS/erc-7590.md:16]: Your abstract should include enough information to give the reader a high-level technical overview of _how_ your proposal works.

File: ERCS/erc-7590.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+## Security Considerations
+
+The same security considerations as with [ERC-721](./eip-721.md) apply: hidden logic may be present in any of the functions, including burn, add resource, accept resource, and more.
+
+Caution is advised when dealing with non-audited contracts.
+
+Implementations MUST use the message sender as from parameter when they are transferring tokens into an NFT. Otherwise, since the current contract needs approval, it could potentially pull the external tokens into a different NFT.
+
+To prevent a seller from front running the sale of an NFT holding [ERC-20](./eip-20.md) tokens to transfer out such tokens before a sale is executed, marketplaces must beware of the `erc20TransferOutNonce` and revert if it has changed since listed.
+
```
Comment (SamWilsn) [inline ERCS/erc-7590.md:175]: I'd also mention that if ERC-20 tokens are transferred directly to the NFT contract, they are lost.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**